zhaquirks/tuya/thermostat_88teujp.py

```diff
@@ -0,0 +1,251 @@
+BATTERY_REPORTED = "battery_reported"
+
+OCCUPIED_HEATING_SETPOINT_COMMAND_ID = 615
+SYSTEM_MODE_COMMAND_ID = 357
+LOCAL_TEMP_COMMAND_ID = 614
+BATTERY_STATE_COMMAND_ID = 1385
+
+OCCUPIED_HEATING_SETPOINT_ATTR = 0x0012
+SYSTEM_MODE_ATTR = 0x001C
+RUNNING_MODE_ATTR = 0x001E
+LOCAL_TEMP_ATTR = 0x0000
+CTRL_SEQ_OF_OPER_ATTR = 0x001B
+BATTERY_PERCENTAGE_REMAINING_ATTR = 0x0021
+
+MIN_HEAT_SETPOINT_ATTR = 0x0015
+MAX_HEAT_SETPOINT_ATTR = 0x0016
+
+
+def payload_to_decimal(data):
+    """Coverts command payload to a single decimal.
+
+    [4, 0, 0, 1, 39] => 295 and [4, 0, 0, 0, 220] => 220
+    """
+    return reduce(lambda acc, i: ((acc << 8) + i) % 2 ** 32, data[1:], 0)
+
+
+def decimal_to_payload(number):
+    """Coverts decimal to command payload.
+
+    e.g. 295 => [4, 0, 0, 1, 39] and 220 => [4, 0, 0, 0, 220]
+    """
+    hex = "{:X}".format(number).rjust(4, "0")
+    chunk1 = int(hex[0:2], 16)
+    chunk2 = int(hex[2:], 16)
+    return [4, 0, 0, chunk1, chunk2]
+
+
+class ManufacturerThermostatCluster(TuyaManufCluster):
+    """Manufacturer thermostat cluster."""
+
+    def handle_cluster_request(self, tsn, command_id, args):
+        """Handle cluster request."""
+        if command_id != TUYA_GET_DATA and command_id != TUYA_SET_DATA_RESPONSE:
+            return
+
+        tuya_cmd = args[0]
+        decimal = payload_to_decimal(tuya_cmd.data)
+
+        if tuya_cmd.command_id == OCCUPIED_HEATING_SETPOINT_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                OCCUPIED_HEATING_SETPOINT_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == LOCAL_TEMP_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                LOCAL_TEMP_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == SYSTEM_MODE_COMMAND_ID:
+            self.endpoint.device.thermostat_bus.listener_event(
+                SYSTEM_MODE_REPORTED, decimal
+            )
+        elif tuya_cmd.command_id == BATTERY_STATE_COMMAND_ID:
+            self.endpoint.device.battery_bus.listener_event(BATTERY_REPORTED, decimal)
+        elif (
+            tuya_cmd.command_id >= 110 and tuya_cmd.command_id <= 129
+        ):  # Ignore historical data and schedules
+            pass
+        elif tuya_cmd.command_id == 362:  # Ignore away mode
+            pass
+        elif tuya_cmd.command_id == 364:  # Ignore auto/manual mode
+            pass
+        else:
+            _LOGGER.debug(
+                "Unknown command received: %d, %d",
+                tuya_cmd.command_id,
+                decimal,
+            )
+
+
+class PowerConfigurationCluster(LocalDataCluster, PowerConfiguration):
+    """Power configuration cluster."""
+
+    cluster_id = PowerConfiguration.cluster_id
+
+    def __init__(self, *args, **kwargs):
+        """Init."""
+        super().__init__(*args, **kwargs)
+        self.endpoint.device.battery_bus.add_listener(self)
+
+    def battery_reported(self, value):
```

Comment:
Another user actually got some real battery status, and I do not really know which devices do provide granular information and which ones do not. I suppose that we could support both.